api/PayrollProcessor.Data.Persistence/Features/Departments/DepartmentEmployeeCreateCommandHandler.cs

```diff
@@ -0,0 +1,35 @@
+using System.Threading;
+using Ardalis.GuardClauses;
+using LanguageExt;
+using Microsoft.Azure.Cosmos;
+using PayrollProcessor.Core.Domain.Features.Departments;
+using PayrollProcessor.Core.Domain.Intrastructure.Operations.Commands;
+
+namespace PayrollProcessor.Data.Persistence.Features.Departments
+{
+    public class DepartmentEmployeeCreateCommandHandler : ICommandHandler<DepartmentEmployeeCreateCommand, DepartmentEmployee>
+    {
+        private readonly CosmosClient client;
+
+        public DepartmentEmployeeCreateCommandHandler(CosmosClient client)
+        {
+            Guard.Against.Null(client, nameof(client));
+
+            this.client = client;
+        }
+
+        public TryOptionAsync<DepartmentEmployee> Execute(DepartmentEmployeeCreateCommand command, CancellationToken token)
+        {
+            var record = DepartmentEmployeeRecord.Map.CreateNewFrom(command.Employee, command.RecordId);
+
+            return async () =>
```

Comment:
`TryOptionAsync<T>` is a representation of a Func that returns a `Task` of a Some<T>/None, though that Func could also completely fail when executed and throw an Exception.

So here we return the callback that when executed could end up in 1 of 3 states

- T - our value returned by executing the callback
- None - no exceptions were thrown but also no values were returned
- Exception - executing the callback caused an exception to be thrown